src/shared/mesh/MeshBuilder.hpp

```diff
@@ -0,0 +1,187 @@
+#ifndef SRC_SHARED_MESHBUILDER_HPP_
+#define SRC_SHARED_MESHBUILDER_HPP_
+
+#include <initializer_list>
+
+#include "mfem.hpp"
+
+#include "tribol/config.hpp"
+
+/**
+ * @file MeshBuilder.hpp
+ * @brief Contains the MeshBuilder and ParMeshBuilder classes to simplify constructing and manipulating MFEM meshes.
+ */
+
+namespace shared {
+
+/**
+ * @class MeshBuilder
+ * @brief A class for building and manipulating MFEM meshes.
+ */
+class MeshBuilder {
+ public:
+  /**
+   * @brief Creates a single mesh from a list of MeshBuilder objects.
+   * @param meshes A list of MeshBuilder objects from which the mesh will be created.
+   * @return A new MeshBuilder object representing the merged mesh.
+   */
+  static MeshBuilder Merged( std::initializer_list<MeshBuilder> meshes );
+
+  /**
+   * @brief Creates a single mesh from multiple MeshBuilder objects.
+   * @tparam Args Variadic template parameter for MeshBuilder objects.
+   * @param meshes MeshBuilder objects from which the mesh will be created.
+   * @return A new MeshBuilder object representing the merged mesh.
+   */
+  template <typename... Args>
+  static MeshBuilder Merged( Args&&... meshes );
+
+  /**
+   * @brief Creates a square mesh occupying the unit square, [0, 1]^2.
+   * @param n_x_els Number of elements in the x direction (>0).
+   * @param n_y_els Number of elements in the y direction (>0).
+   * @return A new MeshBuilder object representing the square mesh.
+   */
+  static MeshBuilder SquareMesh( int n_x_els, int n_y_els );
+
+  /**
+   * @brief Creates a cube mesh occupying the unit cube, [0, 1]^3.
+   * @param n_x_els Number of elements in the x direction (>0).
+   * @param n_y_els Number of elements in the y direction (>0).
+   * @param n_z_els Number of elements in the z direction (>0).
+   * @return A new MeshBuilder object representing the cube mesh.
+   */
+  static MeshBuilder CubeMesh( int n_x_els, int n_y_els, int n_z_els );
+
+  /**
+   * @brief Creates a hypercube mesh occupying the unit hypercube, [0, 1]^dim.
+   * @param dim Dimension of the hypercube (between 1 and 3).
+   * @param n_els Number of elements in each direction of each dimension (>0).
+   * @return A new MeshBuilder object representing the hypercube mesh.
+   */
+  static MeshBuilder HypercubeMesh( int dim, int n_els );
+
+  /**
+   * @brief Constructs a MeshBuilder object from an mfem::Mesh.
+   * @param mesh An rvalue reference to an mfem::Mesh object.
+   */
+  MeshBuilder( mfem::Mesh&& mesh );
+
+  /**
+   * @brief Translates the mesh by a given vector.
```

Comment:
Just confirming that this translates the whole mesh, and does not scale each dimension?

Reply:
Yes, it translates the whole mesh.  I only needed translation for this test.  If other tests/examples need scaling, etc. I can add them in the future.